Tools/cases_generator/stack.py

```diff
@@ -385,18 +386,37 @@ def merge(self, other: "Stack", out: CWriter) -> None:
         self.align(other, out)
 
 
+def stacks(inst: Instruction | PseudoInstruction) -> Iterator[StackEffect]:
+    if isinstance(inst, Instruction):
+        for uop in inst.parts:
+            if isinstance(uop, Uop):
+                yield uop.stack
+    else:
+        assert isinstance(inst, PseudoInstruction)
+        yield inst.stack
+
+
 def get_stack_effect(inst: Instruction | PseudoInstruction) -> Stack:
     stack = Stack()
+    for s in stacks(inst):
+        locals: dict[str, Local] = {}
```

Comment:
That's a chunky bit of copy-paste. Could we instead have an arg to `get_stack_effect` indicating whether we want the intermediate effects?

Otherwise, can this loop be refactored into a shared helper?